src/Reduction.h

```diff
@@ -46,6 +46,26 @@ class ReductionDomain {
 
     /** Immutable access to the reduction variables. */
     EXPORT const std::vector<ReductionVariable> &domain() const;
+
+    /** Add predicate to the reduction domain. See \ref RDom::where
+     * for more details. */
+    EXPORT void where(Expr predicate);
+
+    /** Get the predicates defined on this reducation demain if there is any. */
+    EXPORT const std::vector<Expr> &predicates() const;
+
+    /** Return AND of predicates defined on this reducation domain as one Expr.
+     * If there is no predicate (i.e. all iteration domain in this reduction domain
+     * is valid), this returns an undefined Expr. */
```

Comment:
Two comments here: first, would it make sense/make code easier to return const_true() instead of undefined for an empty predicate list?

Second, consider renaming this to maybe fold_predicates, or just predicate. and_predicates was a little bit ambiguous to me when I first saw it.